packages/image_picker/image_picker/ios/Classes/FLTImagePickerPlugin.m

```diff
@@ -246,12 +298,90 @@ - (void)errorNoPhotoAccess:(PHAuthorizationStatus)status {
   }
 }
 
-- (void)showPhotoLibrary {
+- (void)showPhotoLibrary:(BOOL)pickerFlag {
   // No need to check if SourceType is available. It always is.
-  _imagePickerController.sourceType = UIImagePickerControllerSourceTypePhotoLibrary;
-  [[self viewControllerWithWindow:nil] presentViewController:_imagePickerController
-                                                    animated:YES
-                                                  completion:nil];
+  if (pickerFlag) {
+    [[self viewControllerWithWindow:nil] presentViewController:_pickerViewController
+                                                      animated:YES
+                                                    completion:nil];
+  } else {
+    _imagePickerController.sourceType = UIImagePickerControllerSourceTypePhotoLibrary;
+    [[self viewControllerWithWindow:nil] presentViewController:_imagePickerController
+                                                      animated:YES
+                                                    completion:nil];
+  }
+}
+
+// Limited access to the photo library
+- (void)showLimitedPhotoLibrary API_AVAILABLE(ios(14)) {
+  [[PHPhotoLibrary sharedPhotoLibrary]
+      presentLimitedLibraryPickerFromViewController:_pickerViewController];
+}
+
+- (NSNumber *)getDesiredImageQuality:(NSNumber *)imageQuality {
+  if (![imageQuality isKindOfClass:[NSNumber class]]) {
+    imageQuality = @1;
+  } else if (imageQuality.intValue < 0 || imageQuality.intValue > 100) {
+    imageQuality = [NSNumber numberWithInt:1];
+  } else {
+    imageQuality = @([imageQuality floatValue] / 100);
+  }
+  return imageQuality;
+}
+
+- (void)picker:(PHPickerViewController *)picker
+    didFinishPicking:(NSArray<PHPickerResult *> *)results API_AVAILABLE(ios(14)) {
+  [picker dismissViewControllerAnimated:YES completion:nil];
+  __weak typeof(self) weakSelf = self;
+
+  NSNumber *maxWidth = [weakSelf.arguments objectForKey:@"maxWidth"];
+  NSNumber *maxHeight = [weakSelf.arguments objectForKey:@"maxHeight"];
+  NSNumber *imageQuality = [weakSelf.arguments objectForKey:@"imageQuality"];
+  NSNumber *desiredImageQuality = [weakSelf getDesiredImageQuality:imageQuality];
```

Comment:
nits: These doesn't have to be `weakSelf`. Making them weak might confuse the code reader. Let's just do `self.` for things outside of the block.

Reply:
Ok, done!